File: muscle-map-v1/src/api/ExerciseService.js
import axios from 'axios';

export async function addExercise(exercise){
    return await axios.post(process.env.REACT_APP_API_URL, exercise);
}

export async function getAllExercises(){
    return await axios.get(process.env.REACT_APP_API_URL);
}

export async function getExercise(name = ""){
    return await axios.get(`${process.env.REACT_APP_API_URL}?name=${name}`);
}

File: muscle-map-api/src/main/java/com/muscle_map/muscle_map/exercise/ExerciseService.java
package com.muscle_map.muscle_map.exercise;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class ExerciseService {
    private final ExerciseRepository exerciseRepository;

    @Autowired
    public ExerciseService(ExerciseRepository exerciseRepository) {
        this.exerciseRepository = exerciseRepository;
    }

    public List<Exercise> getAllExercises() {
        return exerciseRepository.findAll();
    }

    public List<Exercise> getExerciseByName(String name) {
        return exerciseRepository.findAll().stream()
                .filter(exercise -> exercise.getName().equalsIgnoreCase(name))
                .collect(Collectors.toList());
    }

    public Exercise addExercise(Exercise exercise) {
        return exerciseRepository.save(exercise);
    }

}
